### DISCUSSION TITLE
Custom report for Behave

### DISCUSSION THREAD

**Discussion by Zubair-codoid** [behave/behave#1130] [Q&A] [answered]
I need to create a custom report for the Behave and Python framework, where I need to fetch the Scenario Name and the scenario status, how to acheive it?

is there any event listener class in the behave framework, in which I can able to call that class and achieve it

Behave version - 1.2.6

**Comment by jenisys** [comment, marked as answer]
Either use an existing `Formatter`, like the JSON Formatter (and a post-processing script that extracts the relevant information from the JSON output) or `JUnitXML` Reporter output.
Otherwise, provide your own `Formatter` (as extension-point in `behave`).

SEE ALSO:
* [features/formatter.user_defined.feature](https://github.com/behave/behave/blob/main/features/formatter.user_defined.feature)

**Reply by Zubair-codoid** [reply]
Thanks @jenisys for the quick response, 
let me try it out and let you know

**Reply by Zubair-codoid** [reply]
@jenisys while executing "behave -f json.pretty features/simple_scenario_with_tags.feature",  JSON output is printed in the terminal.

But i need to fetch that data, is the JSON is stored somewhere in the project location?

**Reply by jenisys** [reply]
Use the `—outfile` option to specify the `output-filename` you want to use. If you do not specify an `output-file` for a `Formatter`, the `Formatter` uses `stdout` as output.

SEE ALSO:

* [features/runner.multiple_formatters.feature](https://github.com/behave/behave/blob/main/features/runner.multiple_formatters.feature)

**Reply by Zubair-codoid** [reply]
Thanks @jenisys

**Comment by jenisys** [comment]
HINT:
* Better use a newer `behave` version from the Git repository, like: [behave v1.2.7.dev5](https://github.com/behave/behave/releases/tag/v1.2.7.dev5)
* It provides better diagnostics, if you make a user error while using an own `Formatter`. For example: `behave -f help` explains you which formatters are available and which have some kind of problem and which problem
